File: python/kylskap.py
p_a, k_a, p_b, k_b, n = map(int, input().split())

best_a, best_b, best_cost = None, None, 3e5
# Notera att vi adderar med två, detta då vi vill testa fallet då vi använder
# endast a_bilar
for a_cars in range(2 + n // k_a):
    for b_cars in range(2 + n // k_b):
        cost = a_cars * p_a + b_cars * p_b
        count = a_cars * k_a + b_cars * k_b
        if n <= count and cost < best_cost:
            best_a = a_cars
            best_b = b_cars
            best_cost = cost

print(best_a, best_b, best_cost)

File: cpp/kylskap.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

#define rep(i, high) for (int i = 0; i < high; i++)
#define repp(i, low, high) for (int i = low; i < high; i++)
#define per(i, high) for (int i = high - 1; i >= 0; i--)
#define perr(i, high, low) for (int i = high - 1, i >= low; i++)
#define sz(container) ((int)container.size())
#define all(x) begin(x), end(x)

void fast() {
  ios::sync_with_stdio(0);
  cin.tie(0);
}

int main() {
  fast();

  ll p_a, k_a, p_b, k_b, n;
  cin >> p_a >> k_a >> p_b >> k_b >> n;

  ll best_a = -1, best_b = -1, best_cost = 3e5;
  // Notera att vi adderar med två, detta då vi vill testa fallet då vi använder
  // endast a_bilar
  for (ll a_cars = 0; a_cars < 2 + n / k_a; a_cars++) {
    for (ll b_cars = 0; b_cars < 2 + n / k_a; b_cars++) {
      ll cost = a_cars * p_a + b_cars * p_b;
      ll count = a_cars * k_a + b_cars * k_b;
      if (n <= count && cost < best_cost) {
        best_a = a_cars;
        best_b = b_cars;
        best_cost = cost;
      }
    }
  }

  cout << best_a << " " << best_b << " " << best_cost << endl;

  return 0;
}
